app/models/miq_worker/runner.rb

```diff
@@ -260,8 +268,8 @@ def do_work_loop
         _log.warn(msg)
         recover_from_temporary_failure
       rescue SystemExit
-        do_exit("SystemExit signal received.  ")
-      rescue => err
+        do_exit("SystemExit signal received.")
+      rescue Exception => err
```

Comment:
So, previously, out of memory or too many open files or whatever would bubble up through here, be uncaught in run single worker and the process would die, logging only what went to STDERR?

My only concern is if we're going to do things in `do_exit` that we should not try to do, like update the worker row, in the case where certain exceptions really need to mean we go down really quickly.

Reply:
For example, if the database connection is dead, we shouldn't try connect to the DB to update a row.

Reply:
@Fryguy ^ please review... am I just being paranoid or should we be careful about doing `do_exit` for crazy exceptions like out of out of memory or errno types.

Reply:
do_exit was designed to be safe for all exit types, including system errors, but I'll double check